evalml/tests/automl_tests/test_training_validation_split.py

```diff
@@ -51,5 +52,15 @@ def test_tvsplit_stratify():
     splitter = TrainingValidationSplit(train_size=5, test_size=5, shuffle=True, stratify=y, random_state=0)
     splits = splitter.split(X, y=y)
     assert len(splits) == 1 and len(splits[0]) == 2
-    pd.testing.assert_index_equal(splits[0][0], pd.Int64Index([1, 4, 2, 8, 7], dtype='int64'))
-    pd.testing.assert_index_equal(splits[0][1], pd.Int64Index([3, 6, 9, 0, 5], dtype='int64'))
+    np.testing.assert_equal(splits[0][0], [1, 4, 2, 8, 7])
+    np.testing.assert_equal(splits[0][1], [3, 6, 9, 0, 5])
+
+
+@pytest.mark.parametrize("random_state", [0, 11, 57, 99, 1000])
+def test_tvsplit_always_within_bounds_with_custom_index(random_state):
+    N = 11000
+    X = pd.DataFrame({'col1': np.arange(0, N)}, index=np.arange(20000, 20000 + N))
+    splitter = TrainingValidationSplit(train_size=0.75, shuffle=True, random_state=random_state)
+    splits = splitter.split(X, y=None)
+    assert np.all(np.logical_and(splits[0][0] < N, splits[0][0] >= 0))
+    assert np.all(np.logical_and(splits[0][1] < N, splits[0][1] >= 0))
```

Comment:
@freddyaboulton I see you defined a custom index here. That makes sense. Shouldn't we check that the output still has the custom indices?
```python
np.testing.assert_equal(20000 + np.arange(0, round(int(0.75*N))), splits[0][0].index)
np.testing.assert_equal(20000 + np.arange(round(int(0.75*N)), N), splits[0][1].index)
```

Reply:
Our splitters don't split the data, they just provide the indices at which the data should be split. But after we call .iloc on these indices, the custom index will be preserved. Should I add that to the test?

Reply:
Ah, you're right, never mind!

Reply:
If I were gonna add another unit test concerning this stuff, it'd be an automl test which runs with `max_pipelines=2` and a dataset with a custom index, mocks pipeline `fit` and makes sure that the data handed to the pipelines is exactly what we expect it to be. Not required for this PR though.